packages/pubsub/src/connection-pool.js

```diff
@@ -459,4 +477,65 @@ ConnectionPool.prototype.shouldReconnect = function(status) {
   return true;
 };
 
+/**
+ *
+ */
+ConnectionPool.prototype.watchForReady = function() {
+  var self = this;
+
+  this.on('newListener', function(eventName) {
+    if (eventName == CHANNEL_READY && !self.waiting) {
+      clearImmediate(self.waitHandle);
+      self.waitHandle = setImmediate(self.waitForReady.bind(self));
+    }
+  });
+};
+
+/**
+ *
+ */
+ConnectionPool.prototype.waitForReady = function() {
+  var self = this;
+  var READY = 2;
+
+  this.getClient(function(err, client) {
+    if (err) {
+      self.emit('error', err);
+      return;
+    }
+
+    var channel = client.getChannel();
+    var state = channel.getConnectivityState(false);
+
+    if (state === READY) {
+      self.removeAllListeners(CHANNEL_ERROR);
+      self.emit(CHANNEL_READY);
+      return;
+    }
+
+    var elapsedTimeWithoutConnection = 0;
+    var deadline, timeout;
```

Comment:
I think we're a multi-line var decl style kinda repo.

Reply:
Yeah, we're not even using `timeout`, I just hadn't gotten around to cleaning up the code yet.